sui_core/src/authority.rs

```diff
@@ -92,6 +96,9 @@ pub struct AuthorityState {
     /// The database
     pub(crate) _database: Arc<AuthorityStore>, // TODO: remove pub
 
+    /// The checkpoint store
+    pub(crate) _checkpoints: Option<Arc<CheckpointStore>>,
```

Comment:
Why `Arc`? Which other task is going to need the CheckpointStore?

Reply:
Also, why `Option`? Only for simplify of integration with the rest of the codebase?

Reply:
here is why, but you might advise me better:
- The consensus will need a pointer to the checkpoint store
- The active logic will need a pointer too

However in both cases maybe its ok to assume we have an Arc to an AuthorityState.

- An option so that we allow testing of the rest of the authority without checkpoints, and also so that the authority can be used by the full node without the checkpointing facility.

Reply:
got it. Let's see when we connect this logic with consensus (we may need a bit more than implementing the trait `ConsensusSender`)